src/auto-evo/simulation/MichePopulation.cs

```diff
@@ -141,157 +163,112 @@ private static List<Species> CopyInitialPopulationsToResults(SimulationConfigura
     }
 
     private static void RunSimulationStep(SimulationConfiguration parameters, List<Species> species,
-        IEnumerable<KeyValuePair<int, Patch>> patchesToSimulate, Random random, SimulationCache cache,
-        IAutoEvoConfiguration autoEvoConfiguration, WorldGenerationSettings worldSettings)
+        IEnumerable<KeyValuePair<int, Patch>> patchesToSimulate, Random random, SimulationCache cache)
     {
         foreach (var entry in patchesToSimulate)
         {
             // Simulate the species in each patch taking into account the already computed populations
             SimulatePatchStep(parameters, entry.Value,
-                species.Where(s => parameters.Results.GetPopulationInPatch(s, entry.Value) > 0),
-                random, cache, autoEvoConfiguration, worldSettings);
+                species.Where(s => parameters.Results.GetPopulationInPatch(s, entry.Value) > 0), random, cache);
         }
     }
 
     /// <summary>
     ///   The heart of the simulation that handles the processed parameters and calculates future populations.
     /// </summary>
     private static void SimulatePatchStep(SimulationConfiguration simulationConfiguration, Patch patch,
-        IEnumerable<Species> genericSpecies, Random random, SimulationCache cache,
-        IAutoEvoConfiguration autoEvoConfiguration, WorldGenerationSettings worldSettings)
+        IEnumerable<Species> genericSpecies, Random random, SimulationCache cache)
     {
         _ = random;
 
         var populations = simulationConfiguration.Results;
         bool trackEnergy = simulationConfiguration.CollectEnergyInformation;
-        bool dayNightCycle = worldSettings.DayNightCycleEnabled;
 
-        // This algorithm version is for microbe species
-        // TODO: add simulation for multicellular
-        var species = genericSpecies.Select(s => s as MicrobeSpecies).Where(s => s != null).Select(s => s!)
-            .ToList();
+        populations.MicheByPatch.TryGetValue(patch, out var miche);
+
+        foreach (var s in simulationConfiguration.ExtraSpecies)
```

Comment:
```suggestion
        foreach (var extraSpecies in simulationConfiguration.ExtraSpecies)
```

Foreach loops can't use single variable names according to our styleguide. Or at least I hope it kind of says that there (with just a few single variable letter names being allowed in some contexts).